src/main/java/net/sigmarik/abilitymod/command/TraitCommand.java

```diff
@@ -87,4 +149,30 @@ private static int printTraits(CommandContext<ServerCommandSource> context) thro
 
         return 1;
     }
+
+    private static class TraitsSuggestionProvider implements SuggestionProvider<ServerCommandSource> {
```

Comment:
Is that... is that a subclass declared in the middle of nowhere, lost in the forest of method definitions?
First I thought you created a completely new class when you used it inside `register()` function.

Subclasses deserve their place on top of the class definition, they are not part of the all structurally insignificant code written inside method definitions.